### PR TITLE
adding jabber protocol (new branch)

### PR DESCRIPTION
**Pull request by nassimabedi**
Adding jabber protocol by new branch.

### CODE REVIEW

File: jabber.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+func readMsgJabber(conn net.Conn, g *Glutton) (err error) {
+	var line []byte
+	r := bufio.NewReader(conn)
+	for i := 1; true; i++ {
+		line, _, err = r.ReadLine()
+		if err != nil {
+			g.logger.Error(fmt.Sprintf("[jabber  ] error: %v", err))
+			return err
+		}
+		dataClient := string(line[:1024])
```
Comment (glaslos) [inline jabber.go]: you cast the `[]byte` to a `string` and then back to `[]byte` in `parseJabberClient`

File: jabber.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	}
+	g.logger.Info(fmt.Sprintf("[jabber  ] STo : %v Version: %v XMLns: %v XMLName: %v", v.STo, v.Version, v.XMLns, v.XMLName))
+	return nil
+}
+
+// read client msg
+func readMsgJabber(conn net.Conn, g *Glutton) (err error) {
+	var line []byte
+	r := bufio.NewReader(conn)
+	for i := 1; true; i++ {
```
Comment (glaslos) [inline jabber.go]: You don't need `i := 1; true; i++`

File: jabber.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+import (
+	"bufio"
+	"context"
+	"encoding/xml"
+	"fmt"
+	"net"
+)
+
+type ServersJabber struct {
```
Comment (glaslos) [inline jabber.go]: Add a doc string here

File: jabber.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+	v := &ServersJabber{Version: "1"}
+	v.Svs = append(v.Svs, serverJabber{"Test_VPN", "127.0.0.1"})
+
+	output, err := xml.MarshalIndent(v, "  ", "    ")
+	if err != nil {
+		g.logger.Error(fmt.Sprintf("[jabber  ]  error: %v", err))
+		return err
+	}
+	conn.Write(output)
```
Comment (glaslos) [inline jabber.go]: this should return an error

File: jabber.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	err = readMsgJabber(conn, g)
```
Comment (glaslos) [inline jabber.go]: You can save a line by doing `if err := readMsgJabber(conn, g); err != nil {`

**Comment by nassimabedi** [comment]
Many Thanks for comments. :)
I think this branch has the same problems in smtp, I close it and create new PR.

**Pull request closed without merging**